TypeCobol.Analysis/Dfa/DefaultDataFlowGraphBuilder.cs

```diff
@@ -32,9 +34,41 @@ private static HashSet<VariableSymbol> GetSymbols(Dictionary<StorageArea, Variab
         public DefaultDataFlowGraphBuilder(ControlFlowGraph<Node, DfaBasicBlockInfo<VariableSymbol>> cfg)
             : base(cfg)
         {
-
         }
 
+        protected override void CollectDataDefinitions()
+        {
+            if (Cfg != null && Cfg.IsInitialized)
+            {
+                var root = Cfg.RootBlock;
+                var dataDivision = this.Cfg.ProcedureDivisionNode.Parent.GetChildren<DataDivision>().FirstOrDefault();
+                if (dataDivision != null)
+                {
+                    foreach(var dataDef in Collect(dataDivision))
+                    {
+                        root.Instructions.AddFirst(dataDef);
+                    }
+                }
+                IEnumerable<DataDefinition> Collect(Node node)
+                {
+                    if (node.SemanticData is VariableSymbol variable && variable.Value != null)
+                    {
+                        yield return node as DataDefinition;
```

Comment:
static cast is ok in fact, so `yield return (DataDefinition) node`.

Reply:
done in 37ed136